### ISSUE TITLE
cmd/compile: checkEnabled causes some debug lines tests to fail

### ISSUE THREAD

**Issue by randall77** [golang/go#54029]
If you set `cmd/compile/internal/ssa/compile.go:checkEnabled` to `true` and run `all.bash`, some tests fail.
```
cmd/compile/internal/ssa/debug_lines_test.go
  TestDebugLinesSayHi
  TestDebugLinesPushback
  TestDebugLinesConvert
cmd/compile/internal/ssa/stmtlines_test.go
  TestStmtLines
```

I suspect this is the randomization done in `cmd/compile/internal/ssa/compile.go` when `checkEnabled` is set.

There is a secondary question here, which is why these tests don't already fail on the ssacheck builder. Maybe the `GO_GCFLAGS` setting doesn't make it through to the subprocess those tests use?

@dr2chase

**Comment by cherrymui** [comment]
> There is a secondary question here, which is why these tests don't already fail on the ssacheck builder. Maybe the GO_GCFLAGS setting doesn't make it through to the subprocess those tests use?

Yeah, it looks like it doesn't pass the flags to subprocesses in those tests.
